src/Xamarin.Android.Build.Tasks/Tasks/AndroidCreateDebugKey.cs

```diff
@@ -27,16 +27,31 @@ protected override CommandLineBuilder CreateCommandLine ()
 			return cmd;
 		}
 
+		bool warnExitCode = false;
+
 		protected override void LogFromStandardError (string text)
 		{
 			// Downgrade error message to warning:
 			// error: java.lang.Exception: Key pair not generated, alias <androiddebugkey> already exists
 			if (text.Contains ($"alias <{KeyAlias}> already exists")) {
 				text = text.Replace ("error:", "warning:");
+				warnExitCode = true;
 			}
 
 			base.LogFromStandardError (text);
 		}
+
+		protected override bool HandleTaskExecutionErrors ()
+		{
+			// Warn for the exit code if we get:
+			// error: java.lang.Exception: Key pair not generated, alias <androiddebugkey> already exists
+			if (warnExitCode && ExitCode != 0) {
```

Comment:
Why check for `ExitCode != 0` when the linked-to `HandleTaskExecutionErrors()` method:

1. has documentation which states:

    > This method will only be called with non-zero exitCode

 2. has a method body which *asserts* that `ExitCode != 0`

Reply:
I removed this check, thanks.